File: java/src/sliding_window/LC_1456_MaximumVowelsSubstringLength.java
package sliding_window;
// https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-given-length


public class LC_1456_MaximumVowelsSubstringLength {

  public int isVowel(char c) {
    return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' ? 1 : 0;
  }

  public int maxVowels(String s, int k) {
    int vowelsMax = 0, vowelsCount = 0;

    for (int i = 0; i < k; i++) {
      vowelsCount += isVowel(s.charAt(i));
    }
    vowelsMax = vowelsCount;
    for (int j = k; j < s.length(); j++) {
      vowelsCount -= isVowel(s.charAt(j - k));
      vowelsCount += isVowel(s.charAt(j));
      vowelsMax = Math.max(vowelsMax, vowelsCount);
    }

    return vowelsMax;
  }
}

File: python/sliding_window/LC_1456_MaximumVowelsSubstringLength.py
# https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-given-length

class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        def isVowel(c):
            return 1 if c in "aeiou" else 0
        vowel_count = 0
        for ch in s[:k]:
            vowel_count += isVowel(ch)
        max_vowel = vowel_count
        for i in range(k, len(s)):
            vowel_count -= isVowel(s[i-k]) 
            vowel_count += isVowel(s[i])
            max_vowel = max(max_vowel, vowel_count)
        return max_vowel